src/main/java/cam72cam/immersiverailroading/tile/TileRail.java

```diff
@@ -56,15 +50,15 @@ public double getMaxRenderDistanceSquared() {
 	}
 
 	public void setSwitchState(SwitchState state) {
-		if (state != info.switchState) {
-			info = new RailInfo(info.world, info.settings, info.placementInfo, info.customInfo, state, info.tablePos);
+		if (state != info.switchState && info.switchForced == SwitchState.NONE) {
```

Comment:
I think we still want to set the switchState here, even if it is being overwritten by switchForced

Reply:
Removed. But if the state passed to setSwitchState() isn't from SwitchUtil.getSwitchState(), there will be issues(aside from initializations of course).